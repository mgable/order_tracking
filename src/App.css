.app {
  margin: 20px;
  min-width: 640px; }
