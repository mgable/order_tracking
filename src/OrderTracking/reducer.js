import {
	ORDER_RECEIVED,
	CANCELLED,
	DELIVERED,
	STATUS_RECEIVED,
	RESET_ORDER,
	TIME_RECEIVED
} from './types';

import { simulationStopped } from '../config';

const initial = {
	raw:[],
	orders: {},
	history: {},
	currentOrder: null,
	status: simulationStopped,
	time: null
};

const Orders = (state = initial, action) => {
	switch (action.type) {
		case ORDER_RECEIVED:
			return orderReceived(state, action);
		case STATUS_RECEIVED:
			return setStatus(state, action)
		case RESET_ORDER:
			return reset(state, action);
		case TIME_RECEIVED:
			return setTime(state, action);
		default:
			return state;
	}
};

const setTime = (state, action) => {
	let time = action.time;
	if (state.time !== time){
		console.info("time is now", time)
		return Object.assign({}, state, {time});
	}

	return state;
}

const reset = (state, action) => {
	return Object.assign({}, {
		raw:[],
		orders: {},
		history: {},
		currentOrder: null,
		status: simulationStopped
	});
}

const setStatus = (state, action) => {
	let status = action.status;
	if (state.status !== status){
		return Object.assign({}, state, {status})
	}

	return state;
}

const cancelOrder = (state, action) => {
	let currentOrder = action.order,
		id = currentOrder.id;
	state.history[id] = currentOrder;
	delete state.orders[id];
	return Object.assign({}, state, {currentOrder})
}

const addOrder = (state, action) => {
	let currentOrder = action.order;
	if (currentOrder.id) {
		state.orders[currentOrder.id] = currentOrder
		state.raw.push(currentOrder);
		return Object.assign({}, state, {currentOrder});
	}

	return state;
}
 
const orderReceived = (state, action) => {
	let currentOrder = action.order;
	switch (currentOrder.event_name) {
		case CANCELLED: 
		case DELIVERED:
			return cancelOrder(state, action);
		default:
			return addOrder(state, action);
	}
};


export default Orders;
