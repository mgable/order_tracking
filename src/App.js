import React from 'react';
import './App.css';
import Orders from './OrderTracking';
import 'bootstrap/dist/css/bootstrap.css';
import {Button} from 'react-bootstrap';
import { Socket, Event } from 'react-socket-io';

 
const uri = 'http://localhost:8000'
const options = { transports: ['websocket'] };

const onClick = () => {
	var socket = window.io(uri);
    socket.emit('system-message', "start");
    console.info("I started");
}

const onMessage = (msg) => {
	console.info(msg);
}
 
export const App = () => {
    return (
    	<div className="app">
    		<Button onClick={onClick}>start</Button>
    		<Socket uri={uri} options={options}> 
                <Event event='system-message' handler={onMessage} />
            </Socket>
        	<Orders />
        </div>
    );
}

export default App;
