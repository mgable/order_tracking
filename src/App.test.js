import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, render } from 'enzyme';
import App from './App';
import configureStore from 'redux-mock-store';


import reducer from './OrderTracking/reducer';

const initialState = {
	orders: {}, // all currently "active" orders by id
	history: {}, // all "delivered" or "cancelled" orders by id
	currentOrder: null, // last order received
	status: "stopped", // simulation status
	time: null, // server time in whole secondes
	threshold: 5, // "cooked" filter threshold time
	serverStatus: null // server socket status
};

const mockStore = configureStore(reducer);
const store = mockStore(initialState)


console.info("store", store)


it('renders without crashing', () => {
    const wrapper = render(<App store={store} />);
  	expect(wrapper).toMatchSnapshot();
});



// import React from 'react';
// import ReactDOM from 'react-dom';
// import { shallow, render } from 'enzyme';


// import createSagaMiddleware from 'redux-saga';
// import configureStore from 'redux-mock-store';

// const sagaMiddleware = createSagaMiddleware();
// const mockStore = configureStore([sagaMiddleware]);
// const store = mockStore(
//   Object.assign(
//     {},
//     {
//       ProductData: { offers: [], upsells: [] },
//       Drinks: {
//         config: {
//           productPicker: { selectConfiguration: { mysteryPack: false } }
//         }
//       }
//     }
//   )
// );

// it('product picker matches correct snapshot', () => {
//   const wrapper = render(<ProductPicker store={store} />);
//   expect(wrapper).toMatchSnapshot();
// });
