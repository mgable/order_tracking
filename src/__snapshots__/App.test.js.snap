// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Order Tracking App renders without crashing 1`] = `
<div
  class="app"
>
  <div
    class="order-tracking container"
  >
    <header>
      <button
        class="status-button stopped btn btn-success"
        type="button"
      >
        Start
      </button>
      <span
        class="simulation-status"
      >
        simulation: stopped
      </span>
      <span
        class="server-status"
      >
        server: 
        <span
          class="circle "
        />
      </span>
      <span
        class="time-threshold"
      >
        <form
          class=""
        >
          <label
            class="form-label"
          >
            cook threshold
          </label>
          <select
            class="form-control"
          >
            <option
              value="5"
            >
              5
            </option>
            <option
              value="10"
            >
              10
            </option>
            <option
              value="15"
            >
              15
            </option>
            <option
              value="20"
            >
              20
            </option>
          </select>
        </form>
      </span>
    </header>
    <div
      class="row panel"
    >
      <div
        class="col-md-6 border panel active"
      >
        <div
          class="status-bar"
        >
          <h3>
            Orders
          </h3>
          <div
            class="dropdown"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              Filter 
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort-down fa-w-10 "
                data-icon="sort-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
        <p>
          There are no orders
        </p>
      </div>
      <div
        class="col-md-6 border panel history"
      >
        <div
          class="status-bar"
        >
          <h3>
            History
          </h3>
          <div
            class="dropdown"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              Filter 
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort-down fa-w-10 "
                data-icon="sort-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
        <p>
          There is no history
        </p>
      </div>
    </div>
  </div>
</div>
`;
